lib/storage/src/content_manager/consensus_manager.rs

```diff
@@ -586,6 +586,38 @@ impl<C: CollectionContainer> ConsensusManager<C> {
         }
     }
 
+    /// Wait and block until consensus reaches a `commit` and `term`
+    ///
+    /// # Returns
+    ///
+    /// Returns `true` if successful.
+    /// Returns `false` on failure, if we have diverged commit/term for example.
+    pub async fn wait_for_consensus_commit(
+        &self,
+        commit: u64,
+        term: u64,
+        consensus_tick: Duration,
+    ) -> bool {
+        // TODO: naive approach with spinlock for waiting on commit/term, find better way
+        loop {
```

Comment:
Do you think it might be better to add explicit timeout here?

Reply:
This is currently only called from a gRPC endpoint. It relies on the request staying alive, when it is dropped, this spin-lock is dropped as well.

In other words, a timeout is currently _enforced_ by the requesting party [here](https://github.com/qdrant/qdrant/blob/f1245640cee3ba78c54897dc48a15b1cc28fe258/lib/storage/src/content_manager/toc.rs#L1688).

The main reason I did this is to not over complicate the possible responses by this gRPC call.

One reason I see to add a timeout here, is to prevent problems if we ever end up using this method directly (locally).

Based on this, do you still have a preference to add a timeout here?

Reply:
the concern is that if loop actually propped once the request if dropped. I assume it is ensured by async call of the timeout?

Reply:
Yeah, when the timeout is reached:
- the gRPC request future is dropped on the client
- the gRPC responding future is dropped on the server
- the spin lock is dropped

I have confirmed this locally.

On second thought, I'll add an explicit timeout anyway to prevent this ever causing problems in the future. :+1:

Reply:
Added an explicit timeout in 2c75afce2261a39b4c8ef3f34795a2545025ba7e